client/client_pool/src/concord_client_pool.cpp

```diff
@@ -473,6 +480,9 @@ void SingleRequestProcessingJob::execute() {
       read_config_.request.max_reply_size = max_reply_size_;
     }
     read_config_.request.reconfiguration = flags_ & RECONFIG_FLAG_REQ;
```

Comment:
is reconfiguration and primary_only mutually exclusive for a single read request?
If yes, then better to place them in if else block like it has been done in above code.